packages/ingest/SftpProviderClient.js

```diff
@@ -19,12 +19,41 @@ class SftpProviderClient {
       this.plaintextPassword = providerConfig.password;
       this.plaintextPrivateKey = providerConfig.plaintextProviderKey;
     }
+
+    this.connected = false;
+    this.privateSftpClient = undefined;
+  }
+
+  async connect() {
+    if (this.connected) return;
+
+    this.privateSftpClient = new SftpClient({
+      host: this.host,
+      port: get(this.providerConfig, 'port', 22),
+      username: await this.getUsername(),
+      password: await this.getPassword(),
+      privateKey: await this.getPrivateKey()
+    });
+
+    await this.privateSftpClient.connect(this.clientOptions);
+
+    this.connected = true;
   }
 
+  async end() {
+    if (this.connected) {
+      this.connected = false;
+
+      await this.privateSftpClient.end();
```

Comment:
Should we do the await before setting this.connected = false in case .end throws?

Reply:
b423db8aa66411a8b8512f7ab02dd6ccb45e1ae9